src/snmp.go

```diff
@@ -44,22 +45,28 @@ func main() {
 	snmpIntegration, err := integration.New(integrationName, integrationVersion, integration.Args(&args))
 	if err != nil {
 		log.Error(err.Error())
-		os.Exit(1)
+		return
+	}
+
+	//log execution time
+	if args.Verbose {
+		startTime := time.Now()
+		defer logExecutionTime(startTime)
 	}
 
 	targetHost = strings.TrimSpace(args.SNMPHost)
 	targetPort = args.SNMPPort
 	err = connect(targetHost, targetPort)
 	if err != nil {
 		log.Error("Error connecting to snmp server "+targetHost, err)
-		os.Exit(1)
+		return
 	}
 	defer disconnect()
 
 	// Ensure a collection file is specified
 	if args.CollectionFiles == "" {
 		log.Error("Must specify at least one collection file")
-		os.Exit(1)
```

Comment:
Suggestion: instead of removing `os.Exit(1)` and returning I would create a function which handles these errors, and I would create one in case it does not exit Something like:
```
handleError(connect(targetHost, targetPort))
...
if args.CollectionFiles == "" {
    handleError(errors.New("Must specify at least one collection file"))
}
```
```
func handleError(err error) {
    log.Error(errors.Wrap(err, msg).Error())
    os.Exit(2)
}
```